ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1252,3 +1252,42 @@ check_parent_attrs_against_processing(enum pkcs11_mechanism_id proc_id,
 
 	return PKCS11_CKR_OK;
 }
+
+bool attribute_is_exportable(struct pkcs11_attribute_head *req_attr,
+			     struct pkcs11_object *obj)
+{
+	uint8_t boolval = 0;
+	uint32_t boolsize = 0;
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	enum pkcs11_class_id key_class = get_class(obj->attributes);
+
+	if (key_class != PKCS11_CKO_SECRET_KEY ||
```

Comment:
bug here :(     should be `&&`, not `||`.

reported by clang thru shippable :)
```
 *  _make COMPILER=clang
ta/pkcs11/src/pkcs11_attributes.c:1264:41: error: overlapping comparisons always evaluate to true [-Werror,-Wtautological-overlap-compare]
        if (key_class != PKCS11_CKO_SECRET_KEY ||
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
1 error generated.

Reply:
Good that tooling caught it. I have fixed the issue.

Reply:
thanks @jforissier :)